src/test/java/picard/sam/SamToFastqTest.java

```diff
@@ -290,25 +275,41 @@ public void testOkGroupedFiles(final String samFilename, final String [] groupFi
         }
     }
 
-    private void runWithOutputPerRg(final String samFilename) {
+    private void runWithOutputPerRg(final String samFilename, final boolean outputPerRg) throws IOException {
         final File samFile = new File(TEST_DATA_DIR, samFilename);
-        final String tmpDir = IOUtil.getDefaultTmpDir().getAbsolutePath() + "/";
-        final String[] args = new String[]{
-                "INPUT=" + samFile.getAbsolutePath(),
-                "OUTPUT_PER_RG=true",
-                "OUTPUT_DIR=" + tmpDir,
-        };
-        runPicardCommandLine(args);
+        final List<String> args = new ArrayList<>(Arrays.asList("INPUT=" + samFile.getAbsolutePath()));
+        if (outputPerRg) {
+            args.add("OUTPUT_DIR=" + IOUtil.getDefaultTmpDir().getAbsolutePath() + "/");
+            args.add("OUTPUT_PER_RG=true");
+            runPicardCommandLine(args);
+        } else {
+            final File pair1File = newTempFastqFile("pair1");
+            final File pair2File = newTempFastqFile("pair2");
+            args.add("FASTQ=" + pair1File.getAbsolutePath());
+            args.add("SECOND_END_FASTQ=" + pair2File.getAbsolutePath());
+            runPicardCommandLine(args);
```

Comment:
extract `runPicardCommandLine(args)` from if/else and call `verifyFastq(pair1File, pair2File, samFile)` in tests where needed.